velocity-tools-view-jsp/src/main/java/org/apache/velocity/tools/view/jsp/jspimpl/VelocityPageContext.java

```diff
@@ -409,20 +408,6 @@ public JspWriter getOut() {
         return jspWriter;
     }
 
-    @SuppressWarnings("deprecation")
-    @Override
-    public javax.servlet.jsp.el.ExpressionEvaluator getExpressionEvaluator() {
-    	// Really, who cares?
-    	throw new UnsupportedOperationException("This class works only with JSP 2.1");
-    }
-
-    @SuppressWarnings("deprecation")
-    @Override
-    public javax.servlet.jsp.el.VariableResolver getVariableResolver() {
-    	// Really, who cares?
-    	throw new UnsupportedOperationException("This class works only with JSP 2.1");
-    }
-
```

Comment:
Can you figure out frim with JSP API version this is not required anymore?

Reply:
I'll try

Reply:
Since 4.0.0, 3.1.1 still has it. See also https://projects.eclipse.org/projects/ee4j.jsp/releases/4.0.0

Reply:
Good, thanks. @arkanovicz I wonder whether we should raise to Servlet 4.0 before that.